test/integration/gems_test.rb

```diff
@@ -113,4 +113,20 @@ class GemsSystemTest < SystemTest
     assert page.has_no_selector?(".gem__users__mfa-text.mfa-warn")
     assert page.has_no_selector?(".gem__users__mfa-text.mfa-info")
   end
+
+  test "shows github link when source_code_uri is set" do
+    github_link = "http://github.com/user/project"
+    create(:version, number: "3.0.1", rubygem: @rubygem, metadata: { "source_code_uri" => github_link })
+
+    visit rubygem_path(@rubygem)
+    assert page.has_selector?(".github-btn")
+  end
+
+  test "shows github link when homepage_uri is set" do
```

Comment:
Is it worth adding a test to validate that there is no Github button when `source_code_uri` and `homepage_uri` is not a URL to github.com?

Reply:
Would you say [this tests](https://github.com/rubygems/rubygems.org/pull/2316/files#diff-93270fdaeb7384163e8c690cdb14e9eaL159-L172) the same thing? let me know if you think this needs integration test as well.

Reply:
added integration test as well.